### ISSUE TITLE
Allow use outside of filters

### ISSUE THREAD

**Issue by sksamuel** [wro4j/wro4j#945]
It's not clear, at least to me, how to use the project outside of servlet filters. I want to use this in a non-servlet environment.

I ended up with something like,

``` scala
  val factory = new SimpleProcessorsFactory()

  factory.addPreProcessor(new CssUrlRewritingProcessor())
  factory.addPreProcessor(new CssImportPreProcessor())
  factory.addPreProcessor(new SemicolonAppenderPreProcessor())
  factory.addPreProcessor(new JawrCssMinifierProcessor())
  factory.addPreProcessor(new JSMinProcessor())
  factory.addPostProcessor(new CssVariablesProcessor())

  val manager = new BaseWroManagerFactory {
    override def newProcessorsFactory(): ProcessorsFactory = factory
  }
```

But after that not sure how to actually apply the processors to say a String. I can see inside the Process method that you're using a Context class to store the request details.

So, if its possible to use this outside of the servlet framework, then please post up a simple example and I will make a PR to add that to the docs.

Or if its not possible, consider this an issue asking for the improvement.

**Comment by alexo** [comment]
Though the initial design was created with servlet-api in mind, it can be used outside the servlet-api (with some tweaks) as well. The best example is the wro4j-maven-plugin (search for AbstractWro4jMojo and Wro4jMojo). 

The only drawback is that you are still required to add the servlet-api artifact dependency, since the wro4j-core is not servlet-api agnostic. On the long term, I would like to get rid of servlet-api dependency from the wro4j-core.

Basically, you have to create a managerFactory and intialize the context (setting the mocked HttpRequest which contains informations about requested group). Once the context is set, you can invoke the **managerFactory.create().process()** method.

If you find this helpful and willing to contribute to documentation, or have suggestions regarding design improvement, you are more than welcome.

**Comment by sksamuel** [comment]
Thank you for the fast response. I've had a further play, and looked at the mojos but I don't think it will be suitable for my purposes. You're using thread locals internally and that's not going to work for my framework (or anything that uses fork join or similar). I notice you had an issue created to address this #421.

I wasn't able to jump in and make any suggestions. I haven't done any Java for a couple of years (this is for a Scala project (https://github.com/scruffy-project/scruffy)) and there's Managers and Factories everywhere that I need to get my head around again :)

**Comment by alexo** [comment]
You could find useful this documentation about [design overview](http://wro4j.readthedocs.org/en/stable/DesignOverview/) where all those concepts are described in detail.

**Comment by sksamuel** [comment]
I get it really, just gently mocking that Java code tends to be full of
Managers, Factories and Impls. The thread local issue is why I can't use
the project as stands, so I'll track the issue created for that.

**Comment by alexo** [comment]
:) .. I'm not offended by this kind of jokes. I agree that sometimes java frameworks are over-engineered.

The threadLocal usage shouldn't be an issue, since it is possible to propagate the Context using [ContextPropagatingCallable](https://github.com/wro4j/wro4j/blob/1.7.x/wro4j-core/src/main/java/ro/isdc/wro/config/support/ContextPropagatingCallable.java).

**Comment by sksamuel** [comment]
What would be ideal is if we can make a ProcessorFactory and then pass that to some Service and then you call service.process(files) or similar. Then I can leverage the existing future setup Scala has.

**Comment by alexo** [comment]
Actually you don't need any ProcessorFactory at all. It is possible to chain processors explicitly. Each processor consumes a stream and writes the processed content to an outputStream. You could iterate through each processor and provide the stream from the files you are trying to process.

**Comment by sksamuel** [comment]
Oh right, let me have a bash at tha tthen.
